django-stubs/utils/decorators.pyi

```diff
@@ -1,15 +1,19 @@
 from collections.abc import Callable, Iterable
 from typing import TypeVar
 
+from django.http.response import HttpResponseBase
 from django.utils.deprecation import MiddlewareMixin
 from django.views.generic.base import View
+from typing_extensions import TypeAlias
 
 _T = TypeVar("_T", bound=View | Callable)  # Any callable
 _CallableType = TypeVar("_CallableType", bound=Callable)
+_VIEW = TypeVar("_VIEW", bound=Callable[..., HttpResponseBase])
+_DECORATOR: TypeAlias = Callable[..., _VIEW | Callable[..., _VIEW]]
```

Comment:
As far as I'm aware, regarding mypy, bounds in `TypeAlias` doesn't work as it's outside of class and function scope. Unfortunately `_DECORATOR` needs to be explicitly expanded to not end up with having `_VIEW` as `Any`.

For `mypy` it should probably be visible if you run a `reveal_type(method_decorator)` somewhere.

Reply:
Here's a playground link that should describe what I'm after: https://mypy-play.net/?mypy=latest&python=3.12&gist=ab4e5d4b4d118bb2e78c146850483952